engine/src/CalciteInterpreter.cpp

```diff
@@ -686,40 +730,51 @@ blazing_frame evaluate_split_query(std::vector<ral::io::data_loader> input_loade
 			}
 		}
 		// these shoudl be split up and run on different threads
-		blazing_frame left_frame;
-		left_frame = evaluate_split_query(input_loaders,
+		std::unique_ptr<ral::frame::BlazingTable> left_frame = evaluate_split_query(input_loaders,
 			schemas,
 			table_names,
 			std::vector<std::string>(query.begin() + 1, query.begin() + other_depth_one_start),
 			queryContext,
 			call_depth + 1);
 
-		blazing_frame right_frame;
-		right_frame = evaluate_split_query(input_loaders,
+		std::unique_ptr<ral::frame::BlazingTable> right_frame = evaluate_split_query(input_loaders,
 			schemas,
 			table_names,
 			std::vector<std::string>(query.begin() + other_depth_one_start, query.end()),
 			queryContext,
 			call_depth + 1);
 
-		blazing_frame result_frame;
 		if(ral::operators::is_join(query[0])) {
 			blazing_timer.reset();  // doing a reset before to not include other calls to evaluate_split_query
 			// we know that left and right are dataframes we want to join together
-			int numLeft = left_frame.get_num_rows_in_table(0);
-			int numRight = right_frame.get_num_rows_in_table(0);
-			left_frame.add_table(right_frame.get_table(0));
+			int numLeft = left_frame->num_rows();
+			int numRight = right_frame->num_rows();
+			
+			// TODO percy felipe william cudf0.12 join why we need to add? could create mem leaks
+			//left_frame.add_table(right_frame.get_table(0));
+			
 			///left_frame.consolidate_tables();
 			std::string new_join_statement, filter_statement;
 			split_inequality_join_into_join_and_filter(query[0], new_join_statement, filter_statement);
-			result_frame = ral::operators::process_join(queryContext, left_frame, new_join_statement);
+			//result_frame = ral::operators::process_join(queryContext, left_frame, new_join_statement);
+			
+			const ral::frame::BlazingTableView table_left = left_frame->toBlazingTableView();
+			const ral::frame::BlazingTableView table_right = right_frame->toBlazingTableView();
+			const std::string expression = new_join_statement;
+			
+			//result_frame = ral::operators::process_join(queryContext, left_frame, new_join_statement);
+			std::unique_ptr<ral::frame::BlazingTable> result_frame = ral::processor::processJoin(table_left, table_right, expression);
```

Comment:
This is rhte non distributed version. Is that the one it is supposed to call? I understand if thats coming in later

Reply:
we are waiting for you to merge that joins stuff

Reply:
thanks